### PR TITLE
Add MISRA 8.13 violation to MISRA.md

### PR DESCRIPTION
**Pull request by yourslab**
Add MISRA 8.13 to MISRA.md

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: platform/posix/transport/src/openssl_posix.c [hunk 1]
```diff
@@ -662,6 +662,7 @@ OpensslStatus_t Openssl_Disconnect( const NetworkContext_t * pNetworkContext )
 }
 /*-----------------------------------------------------------*/
 
+/* coverity[misra_c_2012_rule_8_13_violation] */
```
Comment (sarenameas) [inline platform/posix/transport/src/openssl_posix.c:665]: It is our convention to leaves some specific details right above the violation https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/master/platform/posix/transport/src/plaintext_posix.c#L111
Could you please add those?

File: MISRA.md [hunk 2]
```diff
@@ -28,11 +28,12 @@ Deviations from the MISRA standard are listed below:
 ### Suppressed with Coverity Comments
 | Deviation | Category | Justification |
 | :-: | :-: | :-: |
-| Directive 4.6 | Advisory | Basic numerical types are used in platform implementations, since the POSIX platform functions and openssl functions used in the platform implementations take arguments of these types. |
+| Directive 4.6 | Advisory | Basic numerical types are used in platform implementations, since the POSIX platform functions and OpenSSL functions used in the platform implementations take arguments of these types. |
+| Rule 8.13 | Advisory | The object pointed by `pNetworkContext` is not modified by POSIX sockets or OpenSSL, but other implementations of `TransportRecv_t` and `TransportSend_t` may do so. |
```
Comment (sarenameas) [inline MISRA.md:32]: Let's move this to the "Flagged by Coverity". I tired putting your annotations in the header file and still no luck.

**Review by sarenameas — APPROVED** [review]

**Review by gkwicker — APPROVED** [review]

File: platform/posix/transport/src/openssl_posix.c [hunk 3]
```diff
@@ -662,6 +662,9 @@ OpensslStatus_t Openssl_Disconnect( const NetworkContext_t * pNetworkContext )
 }
 /*-----------------------------------------------------------*/
 
+/* MISRA Rule 18.3 flags the following line for not using the const qualifier
```
Comment (muneebahmed10) [inline platform/posix/transport/src/openssl_posix.c:665]: PR says 8.13, but comment says 18.3, which applies to using relational operators with pointers...

File: MISRA.md [hunk 4]
```diff
@@ -24,15 +24,16 @@ Deviations from the MISRA standard are listed below:
 | Rule 5.6 | Required | The header file `transport_interface.h` is present in each library repository that depends on the transport interface. This is to be able to independently compile the library repository build targets. When all the targets are analyzed together, Coverity flags the type duplication caused by multiple copies of header files in this repository as violations. However, this violation will not be flagged if each library repository build target is analyzed individually. |
 | Rule 5.7 | Required | The network context struct is defined by each transport implementation; however, only one transport implementation is needed by an application at a time. If the transport implementations are analyzed one at a time by Coverity, this violation will not be flagged. |
 | Rule 8.7 | Advisory | API functions are not used by the library outside of the files they are defined; however, they must be externally visible in order to be used by an application. |
+| Rule 8.13 | Advisory | The object pointed by `pNetworkContext` is not modified by POSIX sockets or OpenSSL, but other implementations of `TransportRecv_t` and `TransportSend_t` may do so. |
```
Comment (muneebahmed10) [inline MISRA.md:27]: "pointed **to** by" or "to which `pNetworkContext` points"

**Pull request merged**